CharlieBackend.Business/Services/CurrentUserService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core.Entities;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Security.Claims;
+
+namespace CharlieBackend.Business.Services
+{
+    /// <summary>
+    /// Class that provides access to properties of the current user.
+    /// </summary>
+    public class CurrentUserService : ICurrentUserService
+    {
+        readonly IHttpContextAccessor _httpContextAccessor;
+
+        public CurrentUserService(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public long AccountId
+        {
+            get
+            {
+                string accountIdString = _httpContextAccessor
+                    .HttpContext
+                    .User
+                    ?.Claims
+                    ?.SingleOrDefault(claim => claim.Type == "AccountId")
+                    ?.Value;
+
+                long accountId;
```

Comment:
If this service is gonna be scoped, don't you think that it's better to store properties' data in private members to decrease the number of data evaluation to one? Like that:

```
private long _accountId = -1;

public long AccountId
        {
            get
            {
                 if(_accountId == -1)
                 {
                      ...
                 }
                 else
                 {
                     return _accountId;
                 }
        }
```

Reply:
@andy-dendy-repo 
Optimized getters logic according to your comment in https://github.com/ita-social-projects/WhatBackend/pull/349/commits/21f38d99eecd6eb1980a0d461612db134929e9ae